django_twined/views.py

```diff
@@ -0,0 +1,66 @@
+import json
+
+from django.http import JsonResponse
+from django_twined.models import ServiceRevision
+
+
+def service_revision(request, namespace, name):
+    """Get or create a service revision. If, when getting a service revision, the revision tag isn't provided, the
+    latest revision for that service is returned.
+
+    :param django.core.handlers.wsgi.WSGIRequest request:
+    :param str namespace: the namespace of the service
+    :param str name: the name of the service
+    :return django.http.response.JsonResponse:
+    """
+    if request.method == "GET":
+        # TODO: We can add support for version ranges here later.
+        revision_tag = request.GET.get("revision_tag")
+
+        try:
+            if revision_tag:
+                service_revision = ServiceRevision.objects.get(namespace=namespace, name=name, tag=revision_tag)
+
+            else:
+                service_revision = (
+                    ServiceRevision.objects.filter(
+                        namespace=namespace,
+                        name=name,
+                    )
+                    .order_by("tag")
```

Comment:
Hmm, different people will want to do this differently (also, I'm not sure that order by tag does what you think it does).

I'd suggest defining a TWINED_SERVICE_REVISION_SELECTION_CALLBACK setting, with a default value "django_twined.whateverlocation.select_latest_semantic"
That callback function must return the ServiceRevision instance, given the name and namespace. The default "select_latest_semamtic" can implement the query that you have here (which makes most sense where you simply want the semantic latest, but if you want to control releases, use the default, or do a/b testing, can be overridden).

Reply:
Does ordering by tag not order service revisions by tag alphabetically?

Ok will do

Reply:
I believe that this is alphabetically ordered:
```
10.1.1
2.1.1
```
What we'd be looking for is a multi-column natural sort, with a split on decimals or other characters, and possibly sorted by date for candidate releases (eg where `0.1.1.tom-test-candidate` might happen before `0.1.1.marcus-test-candidate`. But allowing the developer to add their own callback that accepts the filtered queryset, orders it how they want and then returns the instance they want will allow flexibility to do whatever.